legate/core/runtime.py

```diff
@@ -880,7 +1122,197 @@ def dispatch(self, op, redop=None):
         else:
             return op.launch(self.legion_runtime, self.legion_context)
 
-    def _schedule(self, ops):
+
+    def serialize_multiop_metadata(self, numpy_context, ops):
+        """creates a 'header' for a fused op that denotes metadata
+        on each ops inputs, outputs, reductions and scalars
+        """
+        #generate offset maps for all inputs to serialize metadata
+        input_starts, output_starts, offset_starts, offsets= [],[],[],[]
+        reduction_starts, scalar_starts, future_starts, op_ids = [], [], [], []
+        input_start, output_start, offset_start = 0,0,0
+        reduction_start, scalar_start, future_start = 0,0,0
+ 
+        for op in ops:
```

Comment:
I guess we'd eventually want coalescing of arguments here as well. Right now, the fusion code passes the same store used by multiple tasks multiple times, hoping that the launcher at least coalesces region requirements for those duplicates. But, we can identify that the same store is being used in multiple tasks when we scan them for dataflow analysis. So, we could potentially pass a deduplicated list of stores to the fusion task with a mapping between the original indices and those in the store list that is passed to the fusion task.

Reply:
Yeah we could deduplicate these, and I don't imagine this being too difficult. We would just have to store the offset to the dedpulicated store (eg in line 1145) in the `offsets` array, and it should just work